### PR TITLE
Feature/object empty

### PR DESCRIPTION
**Pull request by jonathan-fielding**
Hi

So this is a new function to enable users of the lib to check if a object is empty

Thanks

Jonathan

### CODE REVIEW

**Comment by abozhilov** [comment]
Some suggestions: 

```
function objectEmpty(obj1, obj2 /*, [objn]*/)
``` 

Why you need to check multiple objects for "emptiness"? Also the name is in singular form. I'll suggest you to use `isObjectEmpty` since you return boolean value. 

```
if (obj == null)
```

Why do you treat `undefined`, `null` and other primitives as object? I'd prefer to throw an error instead. 

```
  for(var prop in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, prop)) {
      return false;
    }
  }

  return true;
```

Instead of that code use: 

```
return !Object.keys(obj).length; 
```

**Comment by jonathan-fielding** [comment]
So this is modelled after how lodash's equivalent method for the reason that I want it to be easy for lodash users to migrate over to Just. This is why I do the obj == null thing.

**Comment by angus-c** [comment]
Thanks. I'm ok with not throwing on null/undefined.

I think this would be more useful if it applied to all collections (i.e. Arrays too - eventually could stretch to ES 6 collection types Sets, Maps etc but can leave out for now). So objects checks Object.keys(obj). length, arrays check arr.length, null-like objects return true

I think a better name would be just-is-empty.

Couple if small things: no need to declare the let variable first in the examples - a one liner is just as clear. Also make sure the overall README and the local README examples are in sync.

**Comment by jonathan-fielding** [comment]
Wouldn't this increase complexity, rather than being "just" one thing it's handling lots of things, at this point it becomes a lot closer to lodash and loses its tiny footprint vs lodash equivalent 

I like functions that do one thing, so I know my object is an object I just want to know if it's empty etc

**Comment by angus-c** [comment]
IMO the complexity is minor and the utility gain significant.

**Comment by jonathan-fielding** [comment]
Ok wrote fix but want to ad lots of tests so will finish in morning

**Comment by jonathan-fielding** [comment]
Ok commited all my fixes, would be great if you could review

File: .eslintrc [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "indent": [2, 2],
     "key-spacing": [2, { "beforeColon": false, "afterColon": true }],
     "max-depth": 0,
-    "max-len": [2, 100, 4],
+    "max-len": [2, 120, 4],
```
Comment (angus-c) [inline .eslintrc]: can you revert this please

File: README.md [hunk 2]
```diff
@@ -19,7 +19,8 @@ Guilt-free utilities for every occasion.
 * [Objects](#objects)		
   * [just-extend](#just-extend)		
   * [just-values](#just-values)	  	  
-  * [just-filter-object](#just-filter-object)
+  * [just-filter-object](#just-filter-object)		
+  * [just-is-empty](#just-is-empty)
```
Comment (angus-c) [inline README.md]: move below just-reduce-object

File: packages/is-empty/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+## just-is-empty
+
+Part of a [library](../../../../) of zero-dependency npm modules that do just do one thing.  
+Guilt-free utilities for every occasion.
+
+[Try it now](http://anguscroll.com/just/just-is-empty)
+
+```js
+import isEmpty from 'just-is-empty';
+
```
Comment (angus-c) [inline packages/is-empty/README.md]: no spaces between lines please

File: packages/is-empty/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+isEmpty({a: 3, b: 5}) // false
+
+isEmpty(['a','b']) // false
+
+isEmpty({}) // true
+
+isEmpty([]) // true
+
+isEmpty(null) // true
```
Comment (angus-c) [inline packages/is-empty/README.md]: can you add isEmpty(undefined) example

File: packages/is-empty/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module.exports = isEmpty;
+
+/*
+ isEmpty({a: 3, b: 5}) // false
+
```
Comment (angus-c) [inline packages/is-empty/index.js]: no line spacing

File: packages/is-empty/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ isEmpty(['a','b']) // false
+
+ isEmpty({}) // true
+
+ isEmpty([]) // true
+
+ isEmpty(null) // true
```
Comment (angus-c) [inline packages/is-empty/index.js]: add isEmpty(undefined)

File: packages/is-empty/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+*/
+
+function isEmpty(obj) {
+  return obj == null || (Array.isArray(obj) && !obj.length) || (typeof obj === 'object' && !Object.keys(obj).length);
```
Comment (angus-c) [inline packages/is-empty/index.js]: Can you break these into separate ifs. The reason is that a non empty array will currently unecessarily continue into the third test

if (obj == null) {
  return true
}
if (Array.isArray(obj) {
  return !obj.length
}
if (typeof obj == 'object' {
  return !Object.keys(obj).length;
}

File: packages/is-empty/package.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "just-object-empty",
```
Comment (angus-c) [inline packages/is-empty/package.json]: just-is-empty

File: test/is-empty/index.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+var test = require('tape');
```
Comment (angus-c) [inline test/is-empty/index.js]: please also add tests for 1) empty array, 2) non empty array, 3) undefined and 4) no argument passed

File: README.md [hunk 10]
```diff
@@ -19,7 +19,8 @@ Guilt-free utilities for every occasion.
...
   * [just-map-object](#just-map-object)  		
   * [just-reduce-object](#just-reduce-object)		
   * [just-pick](#just-pick)
```
Comment (angus-c) [inline README.md]: needs main section in README - after just-reduce-object

File: test/is-empty/index.js (hunk 9 above)
Comment (jonathan-fielding) [inline test/is-empty/index.js]: added

**Comment by jonathan-fielding** [comment]
All amends done,

**Comment by angus-c** [comment]
thanks - landed
